### PR TITLE
refactor(other): changes style generation to allow for themes

### PR DESCRIPTION
**Pull request by daragh-king-genesys**
changes style generation to allow for themes

COMUI-2974

### CODE REVIEW

File: packages/genesys-spark-chart-components/src/style/style.scss [hunk 1]
```diff
@@ -2,4 +2,4 @@
  * This file combines all the exported library styles into a single stylesheet for consumption by applications.
  */
 
-@use '~genesys-spark/src/style/global.scss';
+@use '~genesys-spark/dist/scss/global.scss';
```
Comment (daragh-king-genesys) [inline packages/genesys-spark-chart-components/src/style/style.scss:5]: This is the most frequent change in the PR. I added scss files to genesys-spark's distribution so they are available via node_modules and not via the cross monorepo file imports.

**Comment by 321gillian** [comment]
Screenshot 2024-07-17 at 12 38 25
If I try to run this on my local with the following commands, I get an error in the browser console as above. I'm running these from the root the project, not the package itself. The commands all finish ok and there's no errors on the command line program that I can see.  
`git clean -dfx`
`npm ci`
`npm run build`
`npm run dev`

File: packages/genesys-spark/scripts/build-style-sheets.sh [hunk 2]
```diff
@@ -1,4 +1,18 @@
 #!/bin/bash
 
-sass src/style/global.scss dist/global.css
 sass src/style/reset.scss dist/reset.css
+sass src/style/global.scss dist/global.css
+sass src/style/ui.scss dist/ui.css
+
+sass src/style/reset.scss dist/css/reset.css
+sass src/style/global.scss dist/css/global.css
+sass src/style/ui.scss dist/css/ui.css
```
Comment (321gillian) [inline packages/genesys-spark/scripts/build-style-sheets.sh:9]: what's the reason that the files are in both dist/reset and dist/css/reset?

File: packages/genesys-spark/src/index.ts [hunk 3]
```diff
@@ -34,9 +38,11 @@ export function loadSparkFonts(): Promise<void> {
  * application bootstrap if it rejects. Its primary use should be for logging
  * unexpected failures.
  */
-export function registerSparkComponents(): Promise<void> {
+export function registerSparkComponents(opts: registerOptions): Promise<void> {
   const SCRIPT_PATH = 'genesys-webcomponents.esm.js';
-  const STYLE_PATH = 'genesys-webcomponents.css';
+  const STYLE_PATH = opts.theme
```
Comment (daragh-king-genesys) [inline packages/genesys-spark/src/index.ts:43]: @321gillian this is what is causing the error in the console you spotted. I will fix it.

**Review by katie-bobbe-genesys — APPROVED** [review]

**Comment by 321gillian** [comment]
> Screenshot 2024-07-17 at 12 38 25 If I try to run this on my local with the following commands, I get an error in the browser console as above. I'm running these from the root the project, not the package itself. The commands all finish ok and there's no errors on the command line program that I can see. `git clean -dfx` `npm ci` `npm run build` `npm run dev`

This is working fine now

**Review by 321gillian — APPROVED** [review]

File: packages/genesys-spark-components/src/style/flare.scss [hunk 4]
```diff
@@ -0,0 +1,9 @@
+/*
+ * This file combines all the exported library styles into a single stylesheet for consumption by applications.
+ */
+
+@use '~genesys-spark/dist/css/flare/global.css';
+@use '~genesys-spark/dist/css/flare/ui.css';
+@use '~genesys-spark/dist/css/global-focusable.css';
+@use '~genesys-spark/dist/css/global-typography.css';
```
[the remark this answers is not in the document]
Comment (daragh-king-genesys) [reply packages/genesys-spark-components/src/style/flare.scss:8]: Ya that is possible but I imagine unlikely. If it does happen we would create new files that take account of the divergence in the genesys-spark package and then this `flare.scss` file might import something like 
```
@use '~genesys-spark/dist/css/flare/focusable.css';
@use '~genesys-spark/dist/css/flare/typography.css';
```

**Review by gavin-everett-genesys — APPROVED** [review]

**Pull request merged**